onnxruntime/core/providers/cuda/nn/max_pool_with_index.cu

```diff
@@ -44,11 +45,27 @@ __global__ void MaxPoolWithIndexKernel(
   int id = blockIdx.x * blockDim.x + threadIdx.x;
   if (id >= output_size) return;
 
+  auto compute_offset =
+    [height, width, depth, channels](int n_index, int c_index, int h_index, int w_index, int d_index) -> int64_t {
```

Comment:
This function uses NCHWD/NHWDC format.
Some code uses NCDHW or NDHWC as line 57 - 68 below.